### ISSUE TITLE
Class variables must be re-annotated when overridden in a subclass

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1925]
After v1.1.145, subclasses which override class variables must reannotate them because un-annotated variables of a class are considered to be instance variables - even when they are initialised in the class definition.  Therefore, the following:

```py
from typing import ClassVar

class Foo:
    bar: ClassVar[str]

class Baz(Foo):
    bar = ''
```

produces an error: `Instance variable "bar" overrides class variable of same name in class "Foo"`.  This check was also previously present in mypy but was removed in https://github.com/python/mypy/issues/4715 because it reduces the usefulness of `ClassVar`s when specialising a base class.

**Comment by erictraut** [comment]
I'm of two minds on this one.

If your intent is for the variable to remain a `ClassVar` in the subclass, it should probably be explicitly annotated as such within the subclass.

This diagnostic is off by default. If you don't like the way it works, you can leave it disabled.

The fact that mypy changed this behavior is a good argument in favor of changing it, but mypy is much more lax than pyright when it comes to class variables versus instance variables.

We'll discuss this internally and decide whether to change it.

**Comment by erictraut** [comment]
After thinking about this a bit more, I agree that it makes sense to suppress an error in this circumstance. It poses no type safety risk to do so. This change will be in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.146, which I just published. It will also be included in the next release of pylance.
